core/web/job_specs_controller_test.go

```diff
@@ -198,6 +199,47 @@ func TestJobSpecsController_Create_HappyPath(t *testing.T) {
 	assert.Equal(t, httpGet.GetURL(), "https://bitstamp.net/api/ticker/")
 }
 
+func TestJobSpecsController_CreateExternalInitiator_Success(t *testing.T) {
```

Comment:
This test may not be necessary, as the code is covered by core/web/external_initiator_test.go

Reply:
Generally, I think it's worth having both. The integration test will just assert on the larger picture points, and assert on just enough to take the next action. The unit test version can assert on smaller details. The main thing we seem to care about here is the callback firing, but that might change in the future so I'd leave it in.

Reply:
Are the controller tests here considered integration tests?  I thought this was what the tests in /core/internal were for.

Reply:
That's right, integration tests go core/internal. I was just trying to shed some light on why I would prefer to leave this test in, even if it looks very similar to the integration test in `core/internal`.

Reply:
Understood, the point of confusion was the test was generally covered by another unit test.